packages/runtime/src/lib/txn.ts

```diff
@@ -477,3 +486,54 @@ export function calculateFeeCredit(groupTx: EncTx[]): CreditFeeType {
 		requiredFee,
 	};
 }
+
+/**
+ * Retunrs field f of the last inner transaction
+ * @param op ITxna or ITxn opcode
+ * @returns result
+ */
+export function executeITxn(op: ITxna | ITxn): StackElem {
+	const groupTx = op.interpreter.innerTxnGroups[op.interpreter.innerTxnGroups.length - 1];
+	const tx = groupTx[groupTx.length - 1];
+	let result: StackElem;
+	if (
+		op.idx === undefined &&
+		op.interpreter.tealVersion >= 5 &&
+		op.interpreter.mode == ExecutionMode.APPLICATION
+	) {
+		//itxn
+		switch (op.field) {
+			case "NumLogs": {
+				const txReceipt = op.interpreter.runtime.ctx.state.txReceipts.get(tx.txID);
+				const logs: Uint8Array[] = txReceipt?.logs ?? [];
+				result = BigInt(logs.length);
+				break;
+			}
+			case "CreatedAssetID": {
+				result = BigInt(op.interpreter.runtime.ctx.createdAssetID);
+				break;
+			}
+			case "CreatedApplicationID": {
+				result = 0n; // can we create an app in inner-tx?
+				break;
+			}
+			default: {
+				result = txnSpecByField(op.field, tx, [tx], op.interpreter.tealVersion);
+				break;
+			}
+		}
+	} else if (op.idx !== undefined && op.interpreter.tealVersion >= 5 && op.interpreter.mode) {
+		if (op.field === "Logs") {
```

Comment:
Can you explain for me about the logs? I read documents about this but not really get into it